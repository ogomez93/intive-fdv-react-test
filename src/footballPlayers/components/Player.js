import React from 'react';
import moment from 'moment';

export function calculateAge(dob) {
  let date = moment(dob, "YYYY-MM-DD").toDate();
  let today = new Date();
  let timeDiff = Math.abs(today.getTime() - date.getTime());
  let age = Math.ceil(timeDiff / (1000 * 3600 * 24 * 365)) - 1;
  return age;
}

const Player = ({ player }) => (
  <tr>
    <td label="Name" className="border-2 border-black px-1 py-3 sm:py-6">{player.name}</td>
    <td label="Position" className="border-2 border-black px-1 py-3 sm:py-6">{player.position}</td>
    <td label="Nationality" className="border-2 border-black px-1 py-3 sm:py-6">{player.nationality}</td>
    <td label="Age" className="border-2 border-black px-1 py-3 sm:py-6">{calculateAge(player.dateOfBirth)}</td>
  </tr>
);

export default Player;