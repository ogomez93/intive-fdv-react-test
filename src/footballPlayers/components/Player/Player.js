import React from 'react';
import calculateAge from 'footballPlayers/utils/calculateAge';

const Player = ({ dateOfBirth, name, nationality, position }) => (
  <tr>
    <td label="Name" className="border-2 border-black px-1 py-3 sm:py-6">
      {name}
    </td>
    <td label="Position" className="border-2 border-black px-1 py-3 sm:py-6">
      {position}
    </td>
    <td label="Nationality" className="border-2 border-black px-1 py-3 sm:py-6">
      {nationality}
    </td>
    <td label="Age" className="border-2 border-black px-1 py-3 sm:py-6">
      {calculateAge(dateOfBirth)}
    </td>
  </tr>
);

export default Player;
