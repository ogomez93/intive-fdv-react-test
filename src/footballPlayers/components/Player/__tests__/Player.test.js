import React from 'react';
import { shallow } from 'enzyme';

import Player from '../Player';
import props from './fixtures/playerProps';

jest.mock('footballPlayers/utils/calculateAge', () => jest.fn(() => 'mocked age'));

describe('Component: <Player />', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<Player {...props} />);
  });

  it('should render', () =>
    expect(wrapper.isEmptyRender()).toEqual(false));
  
  it('should render correctly', () =>
    expect(wrapper).toMatchSnapshot());

  describe('Player props', () => {
    describe('Name', () => {
      let nameTd;
      
      beforeAll(() => {
        nameTd = wrapper.find('[label="Name"]');
      });

      it('should contain the mocked name', () =>
        expect(nameTd.text()).toEqual('mocked name'));
    });

    describe('Position', () => {
      let positionTd;
      
      beforeAll(() => {
        positionTd = wrapper.find('[label="Position"]');
      });

      it('should contain the mocked position', () =>
        expect(positionTd.text()).toEqual('mocked position'));
    });

    describe('Nationality', () => {
      let nationalityTd;
      
      beforeAll(() => {
        nationalityTd = wrapper.find('[label="Nationality"]');
      });

      it('should contain the mocked nationality', () =>
        expect(nationalityTd.text()).toEqual('mocked nationality'));
    });

    describe('Age', () => {
      let ageTd;
      
      beforeAll(() => {
        ageTd = wrapper.find('[label="Age"]');
      });

      it('should contain the mocked age', () =>
        expect(ageTd.text()).toEqual('mocked age'));
    });
  });
});
