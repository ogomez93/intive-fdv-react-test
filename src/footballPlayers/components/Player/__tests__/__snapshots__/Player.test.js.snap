// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Player /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Player
    dateOfBirth="mocked date of birth"
    name="mocked name"
    nationality="mocked nationality"
    position="mocked position"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td
          className="border-2 border-black px-1 py-3 sm:py-6"
          label="Name"
>
          mocked name
</td>,
        <td
          className="border-2 border-black px-1 py-3 sm:py-6"
          label="Position"
>
          mocked position
</td>,
        <td
          className="border-2 border-black px-1 py-3 sm:py-6"
          label="Nationality"
>
          mocked nationality
</td>,
        <td
          className="border-2 border-black px-1 py-3 sm:py-6"
          label="Age"
>
          mocked age
</td>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "mocked name",
          "className": "border-2 border-black px-1 py-3 sm:py-6",
          "label": "Name",
        },
        "ref": null,
        "rendered": "mocked name",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "mocked position",
          "className": "border-2 border-black px-1 py-3 sm:py-6",
          "label": "Position",
        },
        "ref": null,
        "rendered": "mocked position",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "mocked nationality",
          "className": "border-2 border-black px-1 py-3 sm:py-6",
          "label": "Nationality",
        },
        "ref": null,
        "rendered": "mocked nationality",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "mocked age",
          "className": "border-2 border-black px-1 py-3 sm:py-6",
          "label": "Age",
        },
        "ref": null,
        "rendered": "mocked age",
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td
            className="border-2 border-black px-1 py-3 sm:py-6"
            label="Name"
>
            mocked name
</td>,
          <td
            className="border-2 border-black px-1 py-3 sm:py-6"
            label="Position"
>
            mocked position
</td>,
          <td
            className="border-2 border-black px-1 py-3 sm:py-6"
            label="Nationality"
>
            mocked nationality
</td>,
          <td
            className="border-2 border-black px-1 py-3 sm:py-6"
            label="Age"
>
            mocked age
</td>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "mocked name",
            "className": "border-2 border-black px-1 py-3 sm:py-6",
            "label": "Name",
          },
          "ref": null,
          "rendered": "mocked name",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "mocked position",
            "className": "border-2 border-black px-1 py-3 sm:py-6",
            "label": "Position",
          },
          "ref": null,
          "rendered": "mocked position",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "mocked nationality",
            "className": "border-2 border-black px-1 py-3 sm:py-6",
            "label": "Nationality",
          },
          "ref": null,
          "rendered": "mocked nationality",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "mocked age",
            "className": "border-2 border-black px-1 py-3 sm:py-6",
            "label": "Age",
          },
          "ref": null,
          "rendered": "mocked age",
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
